### PR TITLE
fix Issue 23479 - ImportC recognizes .i and .c files, but not .h files

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by RazvanN7 — APPROVED** [review]

File: compiler/test/compilable/test23479.h [hunk 1]
```diff
@@ -0,0 +1,2 @@
+
+int foo() { }
```
Comment (dkorpel) [inline compiler/test/compilable/test23479.h:2]: This is a useless file currently, the test suite is not going to run this

Comment (WalterBright) [reply compiler/test/compilable/test23479.h:2]: It just needs to compile the .h file.

Comment (dkorpel) [reply compiler/test/compilable/test23479.h:2]: But it doesn't, since you didn't change the test runner to also look for .h files in `test/compilable`

Comment (dkorpel) [reply compiler/test/compilable/test23479.h:2]: https://github.com/dlang/dmd/pull/14640

**Comment by ibuclaw** [comment]
This PR caused a regression. https://issues.dlang.org/show_bug.cgi?id=23547

**Pull request merged**